src/main/java/org/semux/core/Wallet.java

```diff
@@ -379,6 +431,36 @@ public boolean addAccount(Key newKey) throws WalletLockedException {
         }
     }
 
+
+    /**
+     * Adds a new account to the wallet.
+     *
+     * NOTE: you need to call {@link #flush()} to update the wallet on disk.
+     *
+     * @return the new account
+     * @throws WalletLockedException
+     *
+     */
+    public Key addAccount() {
+        requireUnlocked();
+
+        synchronized (accounts) {
+            //will update BIP44 to not throw exception, as we supply all encodings
+            try {
+                HdAddress<HdPrivateKey, HdPublicKey> rootAddress = BIP_44.getRootAddressFromSeed(hdSeed, Network.mainnet, CoinType.semux);
```

Comment:
Why is it parametrized like this? Is this for providing custom subtypes of `HdPrivateKey` and `HdPublicKey`? Is this a useful scenario?

Reply:
It was parameterized when I was doing the alternate HDed25519 implementation that derived the pub/priv keys outside of the main HD key for that level.

As I ended up using SLIP-100, which doesn't require this method, I can remove them.